src/Fixer/Import/OrderedImportsFixer.php

```diff
@@ -187,49 +187,40 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
         $tokensAnalyzer = new TokensAnalyzer($tokens);
         $namespacesImports = $tokensAnalyzer->getImportUseIndexes(true);
 
-        if (0 === \count($namespacesImports)) {
-            return;
-        }
+        foreach (array_reverse($namespacesImports) as $usesPerNamespaceIndices) {
+            $count = \count($usesPerNamespaceIndices);
 
-        $usesOrder = [];
-        foreach ($namespacesImports as $uses) {
-            $usesOrder[] = $this->getNewOrder(array_reverse($uses), $tokens);
-        }
-        $usesOrder = array_replace(...$usesOrder);
+            if (0 === $count) {
+                continue; // nothing to sort
+            }
 
-        $usesOrder = array_reverse($usesOrder, true);
-        $mapStartToEnd = [];
+            if (1 === $count) {
+                $this->setNewOrder($tokens, $this->getNewOrder($usesPerNamespaceIndices, $tokens));
 
-        foreach ($usesOrder as $use) {
-            $mapStartToEnd[$use['startIndex']] = $use['endIndex'];
-        }
+                continue;
+            }
 
-        // Now insert the new tokens, starting from the end
-        foreach ($usesOrder as $index => $use) {
-            $declarationTokens = Tokens::fromCode(
-                sprintf(
-                    '<?php use %s%s;',
-                    self::IMPORT_TYPE_CLASS === $use['importType'] ? '' : ' '.$use['importType'].' ',
-                    $use['namespace']
-                )
-            );
+            $groupUsesOffset = 0;
+            $groupUses = [$groupUsesOffset => [$usesPerNamespaceIndices[0]]];
 
-            $declarationTokens->clearRange(0, 2); // clear `<?php use `
-            $declarationTokens->clearAt(\count($declarationTokens) - 1); // clear `;`
-            $declarationTokens->clearEmptyTokens();
+            // if between two `use`'s is some logic, only sort `use`'s grouped before that logic
+            for ($index = 0; $index < $count - 1; ++$index) {
+                $nextGroupUse = $tokens->getNextTokenOfKind($usesPerNamespaceIndices[$index], [';', [T_CLOSE_TAG]]);
+                $nextGroupUse = $tokens->getNextMeaningfulToken($nextGroupUse);
```

Comment:
I don't like such consequent assignments, it could be:

```suggestion
                $nextGroupUse = $tokens->getNextMeaningfulToken(
                    $tokens->getNextTokenOfKind($usesPerNamespaceIndices[$index], [';', [T_CLOSE_TAG]])
                );
```

Conditional assignment below is OK, but these two are error-prone IMHO.

Reply:
done :+1: